build/dockerfiles/rhel.Dockerfile

```diff
@@ -9,12 +9,11 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-# UPSTREAM: use devtools/go-toolset-rhel7 image so we're not required to authenticate with registry.redhat.io
-# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/devtools/go-toolset-rhel7
-FROM registry.access.redhat.com/devtools/go-toolset-rhel7:1.13.15-4 as builder
-# remember to update path when golang version changes (1.13 -> 1.14)
-ENV PATH=/opt/rh/go-toolset-1.13/root/usr/bin:$PATH \
-    GOPATH=/go/ \
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.4-200.1622548483 as builder
+RUN microdnf install -y golang
```

Comment:
# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8/go-toolset
FROM registry.access.redhat.com/ubi8/go-toolsetl:1.15.7-11 as builder

Reply:
there is typo in an image `toolsetl`

Reply:
fixed

Reply:
I have 2 more issues with this
1. `ubi8/go-toolset` has 1.11GB
2. it has been deprecated and there will be no more updates (https://github.com/eclipse/che/issues/19762)

Reply:
@sparkoo
1. on every run it'll install golang (and then fetch packages)
2 I may be wrong but AFAIK it's rhel7/go-toolset that is deprecated (cc @nickboldt )

Reply:
I've reached out to the go toolset owners for clarity on why in early May 2021 there was a deprecation notice on both the rhel 8.3 and rhel 7 toolset images, but today those warnings are gone.

I still think using the ubi8-minimal image + installing golang rpm will be faster than fetching a 1.1GB image as a builder, so it seems like the current approach is still a good one, and has already been done for jwtproxy, machine exec and che operator. 

OTOH if we DO switch to using the ubi8 go toolset image (1.15 tag), then https://github.com/eclipse/che/issues/19970 becomes a lot simpler -- we could just use that base image for all golang projects, rather than needing a custom che-golang-builder image. cc: @skabashnyuk

Reply:
Go toolset image owners have said that the deprecation right before 8.4 release was a mistake, but it could be deprecated again in future. 

And that "from a size perspective you should certainly do much better with ubi8-minimal + go-toolset"

So... unless there's a compelling reason to NOT use this...

```
FROM registry.access.redhat.com/ubi8-minimal:8.4-200.1622548483 as builder
RUN microdnf install -y golang
```

I would recommend we stick with the plan from last month.

> what about microdnf install golang-1.15.7

Sure, we could lock down a specific version. Or, perhaps we want to inherit that version from some central config file, so that all the che projects can align to the same version of golang?